crates/gpui/src/platform/linux/wayland/client.rs

```diff
@@ -626,6 +626,7 @@ impl Dispatch<wl_pointer::WlPointer, ()> for WaylandClientState {
                 let mut mouse_focused_window = None;
                 for window in &state.windows {
                     if window.1.surface.id() == surface.id() {
+                        window.1.set_focused(true);
```

Comment:
I added a new commit to handle a corner case. For example, in multi-monitor setup, I have zed in one monitor and some other app in another monitor.  When I move the mouse cursor from zed to the other app in another monitor and bring the cursor back on top zed, the cursor icon updates no longer happens until I explicitly click inside the zed app. This appears to be because the window doesn't get the focus back during the `wl_pointer::Event::Enter` event. So, to fix this, I have added code to focus when pointer enter's the window and unfocus when the pointer leave's the window. This already seems to be implemented for `wl_keyboard::Event:Enter` and `wl_keyboard::Event:Leave`. So, I guess this is ok.

Reply:
Wouldn't this cause unintentional "focus on hover" behavior?

That is how I use my desktop(i3/sway), but afaik, it is definitely an unconventional non-default behavior. And causes some issues for some programs (other focused popup window loses focus, and decides to close because of that and such)

Reply:
I also use focus-follows-mouse in KDE Plasma, but it's definitely something you have to go out of your way to enable.

I'm surprised that window focus isn't something managed by the compositor all the way from the core protocol, though?

Reply:
> Wouldn't this cause unintentional "focus on hover" behavior?
That is how I use my desktop(i3/sway), but afaik, it is definitely an unconventional non-default behavior. And causes some issues for some programs (other focused popup window loses focus, and decides to close because of that and such)

I don't see any weird focus behavior with this change. For example, I opened zed in one monitor and another software in second monitor. I right clicked on the another software to open up usual right-click context menu (copy, paste, etc). While it is open, I moved the cursor into zed and the right-click menu doesn't automatically close when I hover over zed. It only get's dismissed when I click inside zed. So, only when i click inside zed the full focus/control gets transferred over to zed. Guess there are two different state like focus and activated or so :thinking: Other apps also exhibit the same behavior. So far no unusual behavior.

Reply:
I see.. It is probably not what I think then.. I will try to test it as well. Thanks!

Reply:
We should definitely double check this.